packages/cli/src/helpers.ts

```diff
@@ -304,3 +306,41 @@ export function toArgs(options: { [key: string]: string | boolean | number | big
     return [flag, stringified];
   });
 }
+
+/**
+ * Extracts the contract and details from the state of a deploy package
+ *
+ * @param state The deploy package state
+ * @returns an object containing ContractData
+ */
+export function getContractsAndDetails(state: { [key: string]: ArtifactContracts }): {
+  [contractName: string]: ContractData;
+} {
+  let contractsAndDetails: { [contractName: string]: ContractData } = {};
+    
+  for (let key in state) {
+      if (key.startsWith('contract.')) {
+          const contracts = state[key]?.artifacts?.contracts;
+          if (contracts) {
+              for (let contractName in contracts) {
+                  contractsAndDetails[contractName] = contracts[contractName];
+              }
+          }
+      }
+  }
+
+  return contractsAndDetails
+}
+
+/**
+ *
+ * @param registries The cannon registries
+ * @returns The LocalRegistry source
+ */
+export function getSourceFromLocalRegistry(registries: CannonRegistry[]): string | undefined {
```

Comment:
should always load from the registry that is configured by the user's settings. the local registry may or may not be in use by a specific user depending on circumstances.